ogr/ogrgeometrycollection.cpp

```diff
@@ -1288,6 +1291,78 @@ double OGRGeometryCollection::get_GeodesicArea(
     return dfArea;
 }
 
+/************************************************************************/
+/*                        get_GeodesicLength()                          */
+/************************************************************************/
+
+/**
+ * \brief Get the length of the collection,where curve edges are geodesic lines
+ * on the underlying ellipsoid of the SRS attached to the geometry.
+ *
+ * The returned length will always be in meters.
+ *
+ * Note that geometries with circular arcs will be linearized in their original
+ * coordinate space first, so the resulting geodesic length will be an
+ * approximation.
+ *
+ * The length is computed as the sum of the lengths of all members
+ * in this collection.
+ *
+ * @note No warning will be issued if a member of the collection does not
+ *       support the get_GeodesicLength method.
+ *
+ * @param poSRSOverride If not null, overrides OGRGeometry::getSpatialReference()
+ * @return the length of the geometry in meters, or a negative value in case
+ * of error.
+ *
+ * @see get_Length() for an alternative method returning areas computed in
+ * 2D Cartesian space.
+ *
+ * @since GDAL 3.10
+ */
+double OGRGeometryCollection::get_GeodesicLength(
+    const OGRSpatialReference *poSRSOverride) const
+{
+    if (!poSRSOverride)
+        poSRSOverride = getSpatialReference();
```

Comment:
For clarity, worth checking for  poSRSOverride not nullptr? I see it's not strictly necessary because it will be checked on the individual items.

Reply:
removed